frontend/packages/ux-editor/src/components/config/editModal/EditOptions/OptionTabs/EditOptionChoice/EditOptionList/OptionListEditor/OptionListEditor.test.tsx

```diff
@@ -0,0 +1,241 @@
+      });
+
+      expect(btnOpen).toBeInTheDocument();
+    });
+
+    it('should open Dialog', async () => {
+      const user = userEvent.setup();
+      await renderOptionListEditorAndWaitForSpinnerToBeRemoved();
+
+      await openOptionModal(user);
+
+      expect(screen.getByRole('dialog')).toBeInTheDocument();
+    });
+
+    it('should close Dialog', async () => {
+      const user = userEvent.setup();
+      await renderOptionListEditorAndWaitForSpinnerToBeRemoved();
+
+      await openOptionModal(user);
+      await user.click(screen.getByRole('button', { name: 'close modal' })); // Todo: Replace "close modal" with defaultDialogProps.closeButtonTitle when https://github.com/digdir/designsystemet/issues/2195 is fixed
+
+      expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
+    });
+
+    it('should call doReloadPreview when editing', async () => {
+      const user = userEvent.setup();
+      const doReloadPreview = jest.fn();
+      await renderOptionListEditorAndWaitForSpinnerToBeRemoved({
+        previewContextProps: { doReloadPreview },
+      });
+
+      await openOptionModal(user);
+      const textBox = screen.getByRole('textbox', {
+        name: textMock('code_list_editor.description_item', { number: 2 }),
+      });
+      await user.type(textBox, 'test');
+      await user.tab();
+
+      await waitFor(() => expect(doReloadPreview).toHaveBeenCalledTimes(1));
+    });
+
+    it('should call updateOptionList with correct parameters when closing Dialog', async () => {
+      const user = userEvent.setup();
+      await renderOptionListEditorAndWaitForSpinnerToBeRemoved();
+      const expectedResultAfterEdit: Option[] = [
+        { value: 'test', label: 'label text', description: 'description', helpText: 'help text' },
+        { value: 2, label: 'label number', description: 'test', helpText: null },
+        { value: true, label: 'label boolean', description: null, helpText: null },
+      ];
+
+      await openOptionModal(user);
+      const textBox = screen.getByRole('textbox', {
+        name: textMock('code_list_editor.description_item', { number: 2 }),
+      });
+      await user.type(textBox, 'test');
+      await user.tab();
+
+      await waitFor(() => expect(queriesMock.updateOptionList).toHaveBeenCalledTimes(1));
+      expect(queriesMock.updateOptionList).toHaveBeenCalledWith(
+        org,
+        app,
+        mockComponent.optionsId,
+        expectedResultAfterEdit,
+      );
+    });
+  });
+});
+
+const openOptionModal = async (user: UserEvent) => {
+  const btnOpen = screen.getByRole('button', {
+    name: textMock('ux_editor.modal_properties_code_list_button_title_library'),
+  });
+  await user.click(btnOpen);
+};
+const openManualModal = async (user: UserEvent) => {
+  const btnOpen = screen.getByRole('button', {
+    name: textMock('ux_editor.modal_properties_code_list_button_title_manual'),
+  });
+  await user.click(btnOpen);
+};
```

Comment:
Naming suggestion:
```suggestion
const openOptionModal = async (user: UserEvent) => {
  const editOptionsButton = screen.getByRole('button', {
    name: textMock('ux_editor.modal_properties_code_list_button_title_library'),
  });
  await user.click(editOptionsButton);
};

const openManualModal = async (user: UserEvent) => {
  const editOptionsButton = screen.getByRole('button', {
    name: textMock('ux_editor.modal_properties_code_list_button_title_manual'),
  });
  await user.click(editOptionsButton);
};
```

Reply:
Edited to not have user.click inside the functions:
```tsx
function getOptionModalButton() {
  return screen.getByRole('button', {
    name: textMock('ux_editor.modal_properties_code_list_button_title_library'),
  });
}

function getManualModalButton() {
  return screen.getByRole('button', {
    name: textMock('ux_editor.modal_properties_code_list_button_title_manual'),
  });
}
```

Reply:
Nice!